pkg/controller/controller.go

```diff
@@ -518,8 +519,8 @@ func (p *csiProvisioner) ProvisionExt(options controller.ProvisionOptions) (*v1.
 	if fsTypesFound > 1 {
 		return nil, controller.ProvisioningFinished, fmt.Errorf("fstype specified in parameters with both \"fstype\" and \"%s\" keys", prefixedFsTypeKey)
 	}
-	if len(fsType) == 0 {
-		fsType = defaultFSType
+	if fsType == "" && p.defaultFSType != "" {
```

Comment:
Can you add unit tests that test the various combinations?
- StorageClass set + default set
- StorageClass unset + default set
- StorageClass set + default unset
- StorageClass unset + default unset

Reply:
Added the test cases ! ..